src/test/java/hudson/plugins/emailext/plugins/content/FailedTestsContentTest.java

```diff
@@ -273,4 +273,98 @@ public void testGetContent_withMessage_withStack_htmlEscaped() {
             "Stack Trace:<br/>" +
             "at org.nexusformat.NexusFile.&lt;clinit&gt;(NexusFile.java:99)<br/><br/>");
     }
+
+    @Test
+    public void testGetContent_withMessage_withStack_specificTestSuit()
+            throws Exception {
+        AbstractTestResultAction<?> testResults = mock( AbstractTestResultAction.class );
+        when( testResults.getFailCount() ).thenReturn( 4 );
+
+        List<TestResult> failedTests = new ArrayList<>();
+
+        for(int i = 0; i < 2; i++) {
+            TestResult result = mock( TestResult.class );
+            when( result.isPassed() ).thenReturn( false );
+            when( result.getFullName() ).thenReturn( "hudson.plugins.emailext.ExtendedEmailPublisherTest" );
+            when( result.getDisplayName() ).thenReturn( "Test" + i );
+            when( result.getErrorDetails() ).thenReturn( "Error" + i );
+            when( result.getErrorStackTrace() ).thenReturn( "Stack" + i );
+            failedTests.add(result);
+        }
+
+        for(int i = 0; i < 2; i++) {
+            TestResult result = mock( TestResult.class );
+            when( result.isPassed() ).thenReturn( false );
+            when( result.getFullName() ).thenReturn( "hudson.plugins.emailext.OtherPackageTest" );
+            when( result.getDisplayName() ).thenReturn( "Test" + i );
+            when( result.getErrorDetails() ).thenReturn( "Error" + i );
+            when( result.getErrorStackTrace() ).thenReturn( "Stack" + i );
+            failedTests.add(result);
+        }
+
+        Mockito.<List<? extends TestResult>>when( testResults.getFailedTests() ).thenReturn( failedTests );
+        when( build.getAction(AbstractTestResultAction.class) ).thenReturn( testResults );
+
+        failedTestContent.maxTests = 4;
+        failedTestContent.showMessage = true;
+        failedTestContent.showStack = true;
+        failedTestContent.testNamePattern = ".*ExtendedEmailPublisherTest.*";
+        String content = failedTestContent.evaluate( build, listener, FailedTestsContent.MACRO_NAME );
+
+        assertTrue( content.contains(2 + " tests failed"));
+        for(int i = 0; i < 2; i++) {
+            assertTrue( content.contains("FAILED:  hudson.plugins.emailext.ExtendedEmailPublisherTest.Test" + i) );
+            assertTrue( content.contains("Error Message:\nError" + i) );
+            assertTrue( content.contains("Stack Trace:\nStack" + i) );
+        }
+        for(int i = 0; i < 2; i++) {
+            assertFalse( content.contains("FAILED:  hudson.plugins.emailext.OtherPackageTest.Test" + i) );
+        }
+    }
+
+    @Test
+    public void testGetContent_withMessage_withStack_allTestSuit()
+            throws Exception {
```

Comment:
I have a couple of comments on this test:
- it is extremely close to the other one, so all the code duplication makes it hard to maintain. However, I do see that you are just following existing pattern.
- The naming (`allTestSuite indicates that it is testing something that returns all suites, however, the reality is that pattern it chose happens to match all. It could as well be just `.*`.